httpcore/_async/connection.py

```diff
@@ -39,6 +39,17 @@ def __init__(
         self.expires_at: Optional[float] = None
         self.backend = AutoBackend()
 
+    def info(self) -> str:
+        if self.connection is None:
+            return "Not connected"
+        elif self.state == ConnectionState.PENDING:
+            return "Connecting"
+        elif self.is_http11:
+            return f"HTTP/1.1, {self.state.name}"
+        else:
+            connection = cast(AsyncHTTP2Connection, self.connection)
+            return f"HTTP/2, {self.state.name}, {len(connection.streams)} streams"
```

Comment:
I'd approach this slightly differently, so...

```python
if self.connection is None:
    return str(self.state.name)
return self.connection.info()
```

And then have `HTTP11Connection` and `HTTP2Connection` implement `.info()`.

I kinda prefer that since it defers the "how do you want to represent the connection info" up to the connection class itself, and doesn't use any casting, which always feels a bit odd.

Reply:
Yup, that makes sense 👍